src/settings/cloud.ts

```diff
@@ -0,0 +1,227 @@
+import { createServer } from 'http';
+import { getConfig, getConfigLocation, setConfigBulk } from '../config';
+import { getGoofCordFolderPath } from "../utils";
+import { shell, dialog } from 'electron';
+import crypto from 'crypto';
+
+import fs from 'fs';
+import path from 'path';
+
+async function cloudConfigCheck() {
+    const filePath = getCloudConfigLocation();
+    try {
+        await fs.promises.access(filePath); 
+    } catch (error) {
+        fs.writeFileSync(filePath, "{}", "utf-8");
+    }
+}
+
+function getCloudConfigLocation(): string {
+    const check = path.join(getGoofCordFolderPath(), "cloud.json");
+    if (!fs.existsSync(check)) {
+        fs.writeFileSync(check, "");
+    }
+    return path.join(getGoofCordFolderPath(), "cloud.json");
+}
+
+async function getCloudToken(): Promise<string | undefined> {
+    const path = getCloudConfigLocation();
+    try {
+        const token = fs.readFileSync(path, "utf-8");
+        console.log("Cloud token:", token);
+        return token;
+    } catch (error) {
+        await cloudConfigCheck();
+        return undefined; 
+    }
+}
+
+function saveCloudToken(token: string) {
+    const path = getCloudConfigLocation();
+    fs.unlinkSync(path);
+    fs.writeFileSync(path, token);
+}
+
+const cloudHost = getConfig("cloudHost");
```

Comment:
getConfig is called before config is initialized, getConfig assumes it's a missing parameter telling setConfig to write to settings, but because cachedConfig is undefined it writes with just "cloudHost" parameter.
Either dynamically import cloud.ts or call getConfig in a function

Reply:
Or as a better solution ./settings/main could be imported dynamically in main.ts

Reply:
I'll add that

Reply:
Alr

Reply:
Decided for a different approach but it works now

Reply:
Great

Reply:
For the "encryption" we might encrypt everything using a default value and if Users want to really encrypt their Settings they can change the encryption key.

Reply:
Maybe. Wait with encryption a bit. I'll do some testing first

Reply:
Okay

Reply:
I've also added you to the cloud repo, the Server automatically updates